qoqo_qiskit/qoqo_qiskit/backend/backend.py

```diff
@@ -0,0 +1,264 @@
+
+        return (
+            output_bit_register_dict,
+            output_float_register_dict,
+            output_complex_register_dict,
+        )
+
+    def run_measurement_registers(
+        self, measurement: Any
+    ) -> Tuple[
+        Dict[str, List[List[bool]]],
+        Dict[str, List[List[float]]],
+        Dict[str, List[List[complex]]],
+    ]:
+        """Run all circuits of a measurement with the PyQuEST backend.
+
+        Args:
+            measurement: The measurement that is run.
+
+        Returns:
+            Tuple[Dict[str, List[List[bool]]],
+                  Dict[str, List[List[float]]],
+                  Dict[str, List[List[complex]]]]
+        """
+        constant_circuit = measurement.constant_circuit()
+        output_bit_register_dict: Dict[str, List[List[bool]]] = dict()
+        output_float_register_dict: Dict[str, List[List[float]]] = dict()
+        output_complex_register_dict: Dict[str, List[List[complex]]] = dict()
+
+        for circuit in measurement.circuits():
+            if constant_circuit is None:
+                run_circuit = circuit
+            else:
+                run_circuit = constant_circuit + circuit
+
+            (
+                tmp_bit_register_dict,
+                tmp_float_register_dict,
+                tmp_complex_register_dict,
+            ) = self.run_circuit(run_circuit)
+
+            output_bit_register_dict.update(tmp_bit_register_dict)
+            output_float_register_dict.update(tmp_float_register_dict)
+            output_complex_register_dict.update(tmp_complex_register_dict)
+
+        return (
+            output_bit_register_dict,
+            output_float_register_dict,
+            output_complex_register_dict,
+        )
+
+    def run_measurement(self, measurement: Any) -> Optional[Dict[str, float]]:
+        """Run a circuit with the PyQuEST backend.
+
+        Args:
+            measurement: The measurement that is run.
+
+        Returns:
+            Optional[Dict[str, float]]
+        """
+        (
+            output_bit_register_dict,
+            output_float_register_dict,
+            output_complex_register_dict,
+        ) = self.run_measurement_registers(measurement)
+
+        return measurement.evaluate(
+            output_bit_register_dict,
+            output_float_register_dict,
+            output_complex_register_dict,
+        )
+
+    def _counts_to_registers(
+        self, counts: List[str]  # result.get_memory() style
+    ) -> Union[List[bool], List[List[bool]]]:
+        bit_map = []
+        reg_num = counts[0].count(" ")
+        for _ in range(reg_num + 1):
+            bit_map.append([])
+        for count in counts:
+            splitted = count.split()
+            for id, measurement in enumerate(splitted):
+                measurement = self._bit_to_bool(measurement)
+                bit_map[id].append(measurement)
+        return bit_map
+
+    def _are_measurement_operations_in(self, input: Circuit) -> bool:
```

Comment:
I think the name of this function is a little confusing. Could it be renamed to `_measurement_operations_in`?